src/psyclone/tests/psyir/transformations/loop_fusion_test.py

```diff
@@ -702,3 +702,62 @@ def test_fuse_no_symbol(fortran_reader, fortran_writer):
         t(ji,jj) = s(ji,jj) + t(ji,jj)
       enddo
     enddo""" in out
+
+
+def test_loop_fuse_different_iterates_over(fortran_reader):
+    ''' Test that an appropriate error is raised when we attempt to
+    fuse two loops that have differing values of ITERATES_OVER '''
+    _, invoke = get_invoke("test11_different_iterates_over_one_invoke.f90",
+                           "gocean1.0", idx=0, dist_mem=False)
+    schedule = invoke.schedule
+    lftrans = LoopFuseTrans()
+
+    # API loops currently only compare the iterates_over attribure
+    with pytest.raises(TransformationError) as err:
+        lftrans.apply(schedule.children[0], schedule.children[1])
+    assert "Loops do not have the same iteration space" in str(err.value)
+
+    # Generic loops compare the loop bounds
```

Comment:
This feels like it should be two tests - one for PSyLoops and one for Loops. Should we have a PSyLoopFusionTrans?

Reply:
I added a TODO to delay this decision. My current understanding is that the property can be computed so the "source of truth" would be the start/stop values, and then the transformation/validate does not need an API specialization. At the moment it looks at iteration_space because in LFRic the boundaries are Literal("NOT_INITIALISED"), so the generic validate currently is insufficient.